### PR TITLE
logger support (see #7)

### PR DESCRIPTION
**Pull request by schillic**
(d6433dc is not related.)

### CODE REVIEW

**Comment by mforets** [comment]
the failing test is because we need to add `Memento.jl` to the list of depdencies. can you try adding it to `REQUIRE`?

File: src/Utils/Utils.jl [hunk 1]
```diff
@@ -218,32 +218,26 @@ INPUT:
 - ``ϕ``    -- a matrix, which can be either dense or sparse
 - ``name`` -- the name of the matrix (for the output message)
 """
-function print_sparsity(ϕ::AbstractVector{Float64}, name::String="")
+function print_sparsity(ϕ::Union{AbstractVector{Float64}, SparseMatrixExp{Float64}}, name::String="")
```
Comment (mforets) [inline src/Utils/Utils.jl]: it's not related to this issue, but shouldn't this be  a `ϕ::Union{AbstractMatrix{Float64}, SparseMatrixExp{Float64}}`?

File: src/logging.jl [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+global G_LOGGER = nothing
+
+const DEFAULT_LOG_LEVEL = "warn"
+
+"""
+    info(msg)
+
+Prints a message on info level.
+
+INPUT:
```
Comment (mforets) [inline src/logging.jl]: please use:

```
### Input

- `msg` -- message string

...
```

File: src/logging.jl [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    configure_logger(level)
+
+Configures the global log level. If no log level is passed, we use the log level
+that is defined by the constant `DEFAULT_LOG_LEVEL`.
+
+INPUT:
+
+- `level` - (optional) the log level; can be either an integer between 0 and 2
+            or a string that is supported by the `Memento.jl` package; see:
+            https://invenia.github.io/Memento.jl/latest/man/intro.html#Logging-levels-1
```
Comment (mforets) [inline src/logging.jl]: you can use markdown here, eg. `[Memento.jl](https://invenia.github.io/Memento.jl/latest/man/intro.html#Logging-levels-1) package`

File: src/options.jl [hunk 4]
```diff
@@ -125,7 +135,9 @@ function validate_solver_options_and_add_default_values!(options::Options)::Opti
 
         # define type/domain constraints for each known key
         domain_constraints = (v  ->  true)
-        if key == :mode
+        if key == :verbosity
+            expected_type = Bool
```
Comment (mforets) [inline src/options.jl]: this return type doesn't correspond to [your comment here](https://github.com/JuliaReach/Reachability.jl/issues/7#issuecomment-342611844)? am i missing something?

OTOH this does work ok for me:

```julia
julia> solve(s, :T=>1.0, :verbosity=>0);

julia> solve(s, :T=>1.0, :verbosity=>1);
[info] Time discretization...
[info] elapsed time: 0.000288061 seconds
[info] Reachable States Computation...
[info] elapsed time: 0.012594997 seconds
[info] Projection...
[info] elapsed time: 0.001423731 seconds

julia> solve(s, :T=>1.0, :verbosity=>2);
[info] Time discretization...
[info] elapsed time: 0.000140385 seconds
[info] Reachable States Computation...
[info] elapsed time: 0.007518403 seconds
[info] Projection...
[info] elapsed time: 0.0013399 seconds
```

**Comment by schillic** [comment]
Thanks, now it worked.
Should we merge?

File: src/options.jl (hunk 4 above)
Comment (mforets) [reply src/options.jl]: but `true/false` doesn't work

File: src/Utils/Utils.jl (hunk 1 above)
Comment (schillic) [reply src/Utils/Utils.jl]: Agreed, let me fix this in `master` directly.

File: src/options.jl (hunk 4 above)
Comment (schillic) [reply src/options.jl]: Great finding, this was old and it was never tested.

File: src/logging.jl (hunk 3 above)
Comment (schillic) [reply src/logging.jl]: Done.

File: src/logging.jl (hunk 2 above)
Comment (schillic) [reply src/logging.jl]: Done.

File: src/Utils/Utils.jl (hunk 1 above)
Comment (schillic) [reply src/Utils/Utils.jl]: Actually, no, there is a second function for this below :)

Comment (mforets) [reply src/Utils/Utils.jl]: we won't have a matrix given as `AbstractVector{Float64}` as input, or if it does it cannot make sense of `ϕ[(2*bi-1):(2*bi), (2*bj-1):(2*bj)]`

Comment (mforets) [reply src/Utils/Utils.jl]: shall we fix it here or outside this issue?

Comment (schillic) [reply src/Utils/Utils.jl]: Do you mean it should be an `AbstractMatrix`?

Comment (mforets) [reply src/Utils/Utils.jl]: yes

Comment (schillic) [reply src/Utils/Utils.jl]: outside

Comment (mforets) [reply src/Utils/Utils.jl]: ok, then let me merge this please.

**Pull request merged**